src/lint/rules/tests/spacing/mod.rs

```diff
@@ -0,0 +1,210 @@
+    // Test rule disable
+    cfg.sp_brace = None;
+    rules = instantiate_rules(&cfg);
+    assert_snippet(SP_BRACES, 0, &rules);
+
+}
+
+static SP_BRACES_02: &str = "
+typedef struct {uint16 idx;} hqm_cq_list_release_ctx_t;
+
+typedef layout \"little-endian\" {bitfields 1 {uint1 cq @ [0:0];} byte;} q_t;
+";
+#[test]
+fn style_check_sp_braces_02() {
+    let mut cfg = LintCfg::default();
+    let mut rules = instantiate_rules(&cfg);
+    assert_snippet(SP_BRACES_02, 6, &rules);
+    // Test rule disable
+    cfg.sp_brace = None;
+    rules = instantiate_rules(&cfg);
+    assert_snippet(SP_BRACES_02, 0, &rules);
+
+}
+
+//  SP.binop around binary operators except the dereferencing operators dot
+//  (a.b) and arrow (a->b)
+#[allow(dead_code)]
+static SP_BINOP: &str = "
+method this_is_some_method() {
+local int this_some_integer = 5+6;
+if (this_some_integer == 0x666)
+    this_some_integer = this.val;
+}
+";
+
+//  SP.ternary around ? and : in the ternary ?: operator
+#[allow(dead_code)]
+static SP_TERNARY: &str = "
+method this_is_some_method(bool flag) {
+local int this_some_integer = (flag?5:7));
+}
+";
+
+//  SP.punct after but not before colon, semicolon and comma
+#[allow(dead_code)]
+static SP_PUNCT: &str = "
+method this_is_some_method(bool flag ,int8 var) {
+    local int this_some_integer = 0x666 ;
+    if(this_some_integer == 0x666)
+        return;
+    some_func(arg1 ,arg2 ,arg3 ,arg4);
+}
+";
+#[test]
+fn style_check_sp_punct_rule() {
+    let mut cfg = LintCfg::default();
+    let mut rules = instantiate_rules(&cfg);
+    assert_snippet(SP_PUNCT, 9, &rules);
+    // Test rule disable
+    cfg.sp_punct = None;
+    rules = instantiate_rules(&cfg);
+    assert_snippet(SP_PUNCT, 0, &rules);
+}
+
+//  SP.ptrdecl between a type and the * marking a pointer
+#[allow(dead_code)]
+static SP_PTRDECL: &str = "
+method this_is_some_method(conf_object_t* dummy_obj) {
+if(!dummy_obj)
+    return;
+}
+";
+
+//  SP.comment around the comment delimiters //, /* and **/
+#[allow(dead_code)]
+static SP_COMMENT: &str = "
+/*Function
+documentation*/
+method this_is_some_method(conf_object_t *dummy_obj) {
+if(!dummy_obj)//Not null
+    return;
+}
+";
+
+// There should be no space:
+//  NSP.funpar between a function/method name and its opening parenthesis
+static NSP_FUNPAR: &str = "
+method this_is_some_method (conf_object_t *dummy_obj) {
+    if(!dummy_obj)
+        other_method_called ();
+}
+";
+#[test]
+fn style_check_nsp_funpar() {
+    let mut cfg = LintCfg::default();
+    let mut rules = instantiate_rules(&cfg);
+    assert_snippet(NSP_FUNPAR, 2, &rules);
+    // Test rule disable
+    cfg.nsp_funpar = None;
+    rules = instantiate_rules(&cfg);
+    assert_snippet(NSP_FUNPAR, 0, &rules);
+}
+
+//  NSP.inparen immediately inside parentheses or brackets
+static NSP_INPAREN: &str = "
+method this_is_some_method( conf_object_t *dummy_obj ) {
+    if( !dummy_obj[ 0 ] )
+        return;
+}
+";
+#[test]
+fn style_check_nsp_inparen() {
```

Comment:
This sort of test is repeated quite a bit, perhaps consider breaking it out into a macro?

Reply:
Sure. I think we can consider this after making the assertions more robust. That might make each test more different from the rest if specific checks need to be made.

Reply:
Yeah, it's a bit of a trade-off between repeating yourself and not being able to change details. Perhaps delay such refactoring until we know more exactly about what we need to test.